app/server.R

```diff
@@ -433,21 +451,23 @@ server <- function(input, output, session) {
       filteredScoreDf <- filterHospitalizationsAheads(filteredScoreDf)
     }
     if (input$scoreType == "wis" || input$scoreType == "sharpness") {
-      # Only show WIS or Sharpness for forecasts that have all intervals
+      # Only show WIS or Sharpness for forecasts that have all intervals unless they are for future dates
       filteredScoreDf <- filteredScoreDf %>%
-        filter(!is.na(`50`)) %>%
-        filter(!is.na(`80`)) %>%
-        filter(!is.na(`95`))
+        filter(!(is.na(`50`) &&
+          is.na(`80`) &&
+          is.na(`95`) &&
```

Comment:
should these be `||` instead of `&&`?

Reply:
Hmmmm.

The original filter was mean to keep only observations (with any `target_end_date`) that had all of the 50, 80, and 95% CIs available. I'm trying to amend the filter to keep only obs that have all of those CIs available OR that have a future `target_end_date`, in which case they won't have the CIs available since the truth data isn't available.

So I want to discard anything that has *any* of those CIs missing *and* has `target_end_date` < the threshold date. To select *in*valid obs,

```
filter((is.na(`50`) ||
          is.na(`80`) ||
          is.na(`95`)) &&
          target_end_date < dataCreationDate)
```

Yep, you're right, those should be `or`s!